cli/azd/pkg/project/remote_build_helper.go

```diff
@@ -0,0 +1,112 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry"
+	"github.com/azure/azure-dev/cli/azd/pkg/containerregistry"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+)
+
+type RemoteBuildHelper struct {
```

Comment:
I spent a fair amount of time sort of chasing my tail here trying to figure out what the factoring should look like.  Originally this logic was on the container app target, but I realized we'd want to re-use it with the AKS target later.  I thought about putting it in `ContainerHelper` (and actually had an iteration that did it) but felt a little strange that that that file was mainly focused around docker and this thing would be orthogonal to the rest of the file.  It would also pull in the RemoteBuildManager and console as new dependencies.

It felt like keeping the seperate was cleaner.  It did mean that `RegistryName` which I wanted to use here needed to move off of `ContainerHelper` but it didn't feel so bad to do that.